### ISSUE TITLE
Filter Option for Status (Downloaded, Missing, Not Monitored, etc.)

### ISSUE THREAD

**Issue by geogolem** [Radarr/Radarr#5004]
**Is your feature request related to a problem? Please describe.**
https://github.com/Radarr/Radarr/issues/4593 was an issue dealing with the fact that the existing filters was not marked/labelled correctly. The PR for that issue correctly names the filter as "Release Status" which better corresponds to the Column Titles in the Options panel.

This issue addresses the lack of "Status" filters on the Movie page.

You can assign this issue to me - I will look into implementing these filters.

**Comment by geogolem** [comment]
Most of the things on the "Status" colum in the Movie Index can be filtered using existing filter options; however, it would be nice to be able to filter the Status column directly as well -- for completeness and ease of use.

**Comment by austinwbest** [comment]
File Status? Movie File Status? On Disk Status? I think just "Status" is still broad since this will be specific to the movie file.

Just my 2 cents :)

**Comment by geogolem** [comment]
I agree about "Status" perhaps being too general --> but I just started looking into this again and I'm still conflicted what really needs to be done here. So lets try and discuss and boil it down to what needs to be done.

The code does refer to the column as "MovieFileStatus" in frontend/src/Movie/MovieFileStatus.js

It makes its decisions using the following fields:

function MovieFileStatus(props) {
  const {
    isAvailable,
    monitored,
    movieFile,
    queueStatus,
    queueState
  } = props;

isAvailable and Monitored are already things available on the custom filters...  I dont think the queuestate stuff is necessary to filter on the movieIndex since the Activity -> Queue should do that IMO...

the only thing that the custom filters is missing is the MovieFile.quality...

**Comment by Fortiori** [comment]
Here mainly to echo the desire to be able to use custom filters on the 'MovieFile.quality' field, however here are some suggestions for alternate names for the "Status" column: (the first two align with the settings tab)
- Quality on Disk
- Quality 
- Downloaded

**Comment by austinwbest** [comment]
> Here mainly to echo the desire to be able to use custom filters on the 'MovieFile.quality' field, however here are some suggestions for alternate names for the "Status" column: (the first two align with the settings tab)
> - Quality on Disk
> - Quality 
> - Downloaded

File properties are not in the plan to be filters.

A missing/downloaded based on a file being present is unlikely to be added as a default since it is a couple clicks for those who want it. 

Size on disk takes care of both of those.

**Comment by anzerman** [comment]
Went looking for a method to filter movies by the quality of existing file - was hoping there was a method to filter via status. 

I still think this would be valuable given that size is only one part of determining quality. 
The length of the film is a pretty big factor to total size as well. 

Ideally I was hoping to filter some of my old DVD quality films and modify just them to look for 1080 bluray for example.

**Comment by pjaclark** [comment]
The queue status is useful in the negative. At the moment, finding size-on-disk=0 will find Missing items, which includes items being downloaded. I want to see things that the automatic search cannot find - ie with a Status of Missing and not Downloading. This is especially an issue when adding a new list, and only some are found by automatic search.

**Comment by bakerboy448** [comment]
> ie with a Status of Missing and not Downloading. This is especially an issue when adding a new list, and only some are found by automatic search.

Do the wanted and missing filters not cover this
Please see the [wiki](https://wiki.servarr.com) Please use one of the support channels: [Reddit](https://www.reddit.com/r/radarr/) or  [Discord](https://radarr.video/discord) for support/questions.

**Comment by pjaclark** [comment]
> Do the wanted and missing filters not cover this

No. Wanted and Missing both include items that are currently downloading (i,e. in table view, the Status column has "Downloading", "Paused" or "Error"; in poster view and overview, they have the colour purple as a bottom border). There is no way to show "missing and not in activity queue" as there is no way to filter on the values in the Status column and no other filter property covers the activity queue.

**Comment by Popupkiller** [comment]
I'd like to add another vote for being able to filter based on quality.
I've tried to do it using size bigger than what the quality I'm looking for is, but this doesn't work, as several movies can have large file sizes even though they are not the quality I'm looking for.
For instance, I'd like to filter only 4K material, so that I can easily bulk edit/move them. but this is not possible, so I have to manually scroll through the entire list, and select each and every item, or reversely, deselect the ones I don't want, which is just as bad.

**Comment by bakerboy448** [comment]
Filter based on current file quality will likely not occur

**Comment by DuvelCorp** [comment]
The only filter that I really miss is downloaded or not downloaded. I like to purge Radarr from time to tome, for everything that doesn't need to be there anymore.

**Comment by bakerboy448** [comment]
> The only filter that I really miss is "**downloaded**". I like to purge Radarr from time to tome, for everything that doesn't need to be there anymore.

Custom Filter
  Size on Disk > 0

Done.

**Comment by lefoy** [comment]
I would also like a way to filter out movies that are already in the download queue

**Comment by tastyratz** [comment]
I understand it was mentioned that filtering is unlikely on downloaded status, however, I wanted another comment to the usefulness of this. It is not intuitive to not be able to filter existing displayed columns.
I keep my 4k content in a different location for a different library. Being able to sort by what the files actually are is helpful, especially when considering old releases that were never made available in 4k online and getting more obscure by the day so they may never show up in 4k. 

Filtering by cutoff unmet is far too broad of a catch considering the number of steps without a resolution cutoff unmet in this scenario.  Sorting by size doesn't work when you consider some 4k releases can be 2gb h.265 files. 

For me, if it were ever possible to filter based on what shows up in the "status" column when selected it would be helpful.

**Comment by averon420** [comment]
I would really love, if that would be added. Already ran two times into the situation, where I needed this. Also the pre-defined filter kinda shows, that its technically already implemented. Just not accessable by users. :) +1 for this

**Comment by Mansour-J** [comment]
This worked well for me!

![image](https://user-images.githubusercontent.com/12991664/188318310-5be8a8d0-6cc3-422b-bdbf-440a77c63da6.png)

**Comment by dandirkmn** [comment]
Not being able to filter on a default displayed column is sort of unexpected.

Looks like what is displayed if a file exists/discovered is the "Quality" used to determine size allowed and to configure quality profiles.

I was trying to find all SD content WEBDL-480p for instance, and this was the first thing I attempted.

**Comment by katbyte** [comment]
Another vote for being able to filter on "status":
<img width="362" alt="image" src="https://user-images.githubusercontent.com/1638467/203417502-8125e925-4de2-4034-94af-162ad682b10b.png">

it seems weird that this column is right there and we cannot filter on it to only show movies of a certain quality.

what is the reason for not supporting this as the data is right there?

**Comment by bakerboy448** [comment]
Additional comments saying this is wanted is not necessary and adds no value.

This likely also won't be done with v4 given it'd need to be completely rewrote in v5 and multi-file support.

**Comment by planetrocky** [comment]
> Another vote for being able to filter on "status": <img alt="image" width="362" src="https://user-images.githubusercontent.com/1638467/203417502-8125e925-4de2-4034-94af-162ad682b10b.png">
> 
> it seems weird that this column is right there and we cannot filter on it to only show movies of a certain quality.
> 
> what is the reason for not supporting this as the data is right there?

Google brought here.  I was trying to find a way to filter which movies I have now got in UHD - not possible from radarr GUI alas.

**Comment by KaiStarkk** [comment]
The data is there... Does anyone in comments have time to write a userscript to filter based on Status?

**Comment by codewatzen** [comment]
Being able to filter off of these
![image](https://user-images.githubusercontent.com/25915728/217853464-a4d3280a-1c1d-4c76-a244-0a9ccb69b46c.png)
would be fantastic. 

Especially since trying to make a filter that shows Downloaded and Monitored only gives me 2 out of the about 40 I have is frustrating. 
![image](https://user-images.githubusercontent.com/25915728/217853879-37035514-e1a2-49f7-b92f-80e58caa765c.png)

**Comment by codewatzen** [comment]
> Being able to filter off of these ![image](https://user-images.githubusercontent.com/25915728/217853464-a4d3280a-1c1d-4c76-a244-0a9ccb69b46c.png) would be fantastic.
> 
> Especially since trying to make a filter that shows Downloaded and Monitored only gives me 2 out of the about 40 I have is frustrating. ![image](https://user-images.githubusercontent.com/25915728/217853879-37035514-e1a2-49f7-b92f-80e58caa765c.png)

I did get the Downloaded (monitored) custom filter to work with this. 
![image](https://user-images.githubusercontent.com/25915728/217856777-c379949e-ea54-4fb5-ac3a-d0710f669d73.png)

**Comment by ManiMatter** [comment]
Also vote for being able to _SORT_ by the status:

- Downloaded (Monitored)
- Downloaded (Unmonitored)
- Missing (Monitored)
- Missing (Unmonitored)
- Queued
- Unavailable

Also vote for being able to click on the above labels to _FILTER_

**Comment by bakerboy448** [comment]
> I hear it mentioned that this issue is not likely be prioritized? May I ask why?

from earlier:

> This likely also won't be done with v4 given it'd need to be completely rewrote in v5 and multi-file support.

**Comment by edrock200** [comment]
Was about to open a feature request but my search for existing led me here. I see cut-off unmet is a built in filter, but is not an available option for custom filters, unless I'm missing it. I'm trying to make a "cut-off met" filter, but don't see a way of doing this. Is this possible? Thanks in advance.

**Comment by bakerboy448** [comment]
Please use one of the support channels: [Reddit](https://www.reddit.com/r/radarr/) or  [Discord](https://radarr.video/discord) for support/questions.

**Comment by LouwMarais** [comment]
I raised a duplicate and further searches led me here. 
duplicate closed: https://github.com/Radarr/Radarr/issues/9276

Any update on how close this is to be added to 5.x ?

**Comment by gmcouto** [comment]
This would be of great help. From time to time I have movies stuck on `Error` or `Downloading`, and I need to import them manually.

It would be nice to go through only them... instead of all missing movies.

![Image](https://github.com/user-attachments/assets/fe589e97-fe6c-4e03-97a0-6612171da5c4)
